src/cli/index.js

```diff
@@ -160,7 +140,7 @@ if (command && typeof command.setFlags === 'function') {
   command.setFlags(commander);
 }
 
-if (commandName === 'help' || args.indexOf('--help') >= 0 || args.indexOf('-h') >= 0) {
+if (args.indexOf('--help') >= 0 || args.indexOf('-h') >= 0) {
```

Comment:
This if now covers only the 2nd case, before it was covering the the 1st, 2nd and the 3rd.